datasets/drop/drop.py

```diff
@@ -69,28 +89,75 @@ def _split_generators(self, dl_manager):
             datasets.SplitGenerator(
                 name=datasets.Split.TRAIN,
                 # These kwargs will be passed to _generate_examples
-                gen_kwargs={"filepath": os.path.join(data_dir, "drop_dataset_train.json")},
+                gen_kwargs={"filepath": os.path.join(data_dir, "drop_dataset_train.json"), "split": "train"},
             ),
             datasets.SplitGenerator(
                 name=datasets.Split.VALIDATION,
                 # These kwargs will be passed to _generate_examples
-                gen_kwargs={"filepath": os.path.join(data_dir, "drop_dataset_dev.json")},
+                gen_kwargs={"filepath": os.path.join(data_dir, "drop_dataset_dev.json"), "split": "validation"},
             ),
         ]
 
-    def _generate_examples(self, filepath):
+    def _generate_examples(self, filepath, split):
         """Yields examples."""
         # TODO(drop): Yields (key, example) tuples from the dataset
         with open(filepath, encoding="utf-8") as f:
             data = json.load(f)
-            for i, key in enumerate(data):
-                example = data[key]
-                qa_pairs = example["qa_pairs"]
-                for j, qa in enumerate(qa_pairs):
-                    question = qa["question"]
-                    answers = qa["answer"]["spans"]
-                    yield str(i) + "_" + str(j), {
-                        "passage": example["passage"],
-                        "question": question,
-                        "answers_spans": {"spans": answers},
+            for i, (section_id, section) in enumerate(data.items()):
+                for j, qa in enumerate(section["qa_pairs"]):
+
+                    example = {
+                        "section_id": section_id,
+                        "query_id": qa["query_id"],
+                        "passage": section["passage"],
+                        "question": qa["question"],
                     }
+
+                    if split == "train":
+                        answers = [qa["answer"]]
+                    else:
+                        answers = qa["validated_answers"]
+
+                    try:
+                        example["answers_spans"] = self.build_answers(answers)
+                        yield example["query_id"], example
+                    except AssertionError:
```

Comment:
Maybe you can have a more explicit error type here ?
Feel free to create a `AnswerParsingError` exception instead of using `assert`.